### PR TITLE
Minor fixes for parse.py

### LINKED ISSUE #15: Update condition check
* Can the `parsed_info` be None and `child_parsed_info` be non None?
* Since this is `and`, we can move check for `parsed_info` outside the loop
* Since `parsed_info` depends only on `cursor.spelling`, we can have an early exit, resulting in out-denting of lines 83-98 (incoming)

_Originally posted by @kunaltyagi in https://github.com/divmadan/pcl/pull/10_

***

> > Can the parsed_info be None and child_parsed_info be non None?
> 
> Yes for now, but that's an error that needs resolving #11. Will update after #11. (The is cursor.spelling was used to ignore empty symbols, but the unnamed structs, enums, etc. requires special treatment)
> 
> > Since this is and, we can move check for parsed_info outside the loop
> > Since parsed_info depends only on cursor.spelling, we can have an early exit, resulting in out-denting of lines 83-98 (incoming)
> 
> Updation after #11

### LINKED ISSUE #17: [new] Add get_includes method to parse.py
How to do:
- Parse using `options=clang.TranslationUnit.PARSE_DETAILED_PROCESSING_RECORD`
- Function:
```py
def get_includes(node):
    includes = []

    for child_node in valid_children(node):
        child_cursor = child_node["cursor"]
        
        if child_cursor.kind.name == "INCLUSION_DIRECTIVE":
            includes.append(child_cursor.displayname)

    return includes
```

### PR DESCRIPTION
**Pull request by diivm**
- Resolves #17 
- Resolves #15 
- Check for `is_node_in_this_file` shifted to `valid_children`

### CODE REVIEW

File: bindings/python/scripts/parse.py
Comment (kunaltyagi) [inline bindings/python/scripts/parse.py:192]: Why this option?

**Review by kunaltyagi — APPROVED** [review]

File: bindings/python/scripts/parse.py
Comment (aPonza) [reply bindings/python/scripts/parse.py:192]: from #17 seems like it would add the `INCLUSION_DIRECTIVE`-values for `cursor.kind.name`, but then a single-shot PR to solve #17 completely would've made more sense

EDIT: [cindex.py isn't more helpful](https://github.com/llvm-mirror/clang/blob/master/bindings/python/clang/cindex.py#L2734-L2736)

Comment (kunaltyagi) [reply bindings/python/scripts/parse.py:192]: @divmadan Please document the reason in code comments

Comment (diivm) [reply bindings/python/scripts/parse.py:192]: Required to get the `INCLUSION_DIRECTIVE`s (included filenames in the source file) Without this option, clang doesn't retain that information.

From the C API:
```
/**                                                                         
 * \brief Used to indicate that the parser should construct a "detailed"    
 * preprocessing record, including all macro definitions and instantiations.
 *                                                                          
 * Constructing a detailed preprocessing record requires more memory        
 * and time to parse, since the information contained in the record         
 * is usually not retained. However, it can be useful for                   
 * applications that require more detailed information about the            
 * behavior of the preprocessor.                                            
 */                                                                         
CXTranslationUnit_DetailedPreprocessingRecord = 0x01,
```

***

> document the reason in code comments

Done :+1:

File: bindings/python/scripts/parse.py
Comment (kunaltyagi) [inline bindings/python/scripts/parse.py]: What about just `child.location.file` for truthiness? It's stronger in the sense that file as empty string aka `''` will also not be allowed

File: bindings/python/tests/test_parse.py
Comment (kunaltyagi) [inline bindings/python/tests/test_parse.py:91]: Nice

**Review by kunaltyagi — APPROVED** [review]
LGTM for merging

Please wait on @aPonza 's review

**Review by aPonza — APPROVED** [review]
Forgot to add the LGTM approval after the comment

**Pull request merged**
